contracts/src/integrations/cert/EigenDACertVerifier.sol

```diff
@@ -46,22 +72,76 @@ contract EigenDACertVerifier is
         _eigenDASignatureVerifier = initEigenDASignatureVerifier;
         _securityThresholds = initSecurityThresholds;
         _quorumNumbersRequired = initQuorumNumbersRequired;
     }
 
+    /// @notice Decodes a certificate from bytes to an EigenDACertV3
+    /// @dev This function is external for the purpose of try/catch'ing it inside checkDACert.
+    function decodeCert(bytes calldata data) external pure returns (CT.EigenDACertV3 memory cert) {
+        return abi.decode(data, (CT.EigenDACertV3));
+    }
+
     /// @inheritdoc IEigenDACertVerifierBase
+    /// @dev This function try catches checkDACertReverts, and maps any reverts to status codes.
+    /// @dev Make sure to call this at a block number that is > RBN, otherwise this function will
+    /// return an INVALID_CERT status code because of a require in the BLSSignatureChecker library that we use.
+    /// TODO: we should return (uint8, bytes) instead and include the revert reason.
     function checkDACert(bytes calldata abiEncodedCert) external view returns (uint8) {
-        (CertLib.StatusCode status,) = CertLib.checkDACert(
-            _eigenDAThresholdRegistry,
-            _eigenDASignatureVerifier,
-            abiEncodedCert,
-            _securityThresholds,
-            _quorumNumbersRequired
+        CT.EigenDACertV3 memory daCert;
+        // We try catch this here because decoding error would appear as a Panic,
+        // which we consider bugs in the try/catch for the checkDACertReverts call below.
+        try this.decodeCert(abiEncodedCert) returns (CT.EigenDACertV3 memory _daCert) {
+            daCert = _daCert;
+        } catch {
+            return uint8(StatusCode.INVALID_CERT);
+        }
+
+        // The try catch below is used to filter certs into 3 status codes:
+        // 1. success
+        // 2. invalid cert (any failing require statement; we assume all require statements return either a string or custom error)
+        // 3. bug (everything else, including solidity panics and low-level evm reverts)
+        try this.checkDACertReverts(daCert) {
+            return uint8(StatusCode.SUCCESS);
+        } catch Error(string memory) /*reason*/ {
+            // This matches any require(..., "string reason") revert that is pre custom errors,
+            // which many of our current eigenlayer-middleware dependencies like the BLSSignatureChecker still use. See:
+            // https://github.com/Layr-Labs/eigenlayer-middleware/blob/fe5834371caed60c1d26ab62b5519b0cbdcb42fa/src/BLSSignatureChecker.sol#L96
+            return uint8(StatusCode.INVALID_CERT);
+        } catch Panic(uint256) /*errorCode*/ {
+            // This matches any panic (e.g. arithmetic overflow, division by zero, invalid array access, etc.),
+            // which means a bug or misconfiguration of the CertVerifier contract itself.
+            return uint8(StatusCode.BUG);
+        } catch (bytes memory reason) {
+            if (reason.length == 0) {
```

Comment:
is this conditional still necessary?

Reply:
no but I added it to make it cleaner. Since length==0 maps to exact case thatn I know of, whereas length=1..4 I have no idea whether its even possible.

Reply:
roger that makes sense